core/block_svg.ts

```diff
@@ -603,10 +609,23 @@ export class BlockSvg
    *
    * @param e Pointer down event.
    */
-  private onMouseDown_(e: PointerEvent) {
+  private onPointerdown_(e: PointerEvent) {
+    const gesture = this.workspace.getGesture(e);
+    if (gesture) {
+      gesture.handleBlockStart(e, this);
+    }
+  }
+
+  /**
+   * Handle a contextMenu on an SVG block.
+   *
+   * @param e Pointer contextmenu event.
+   */
+  private onContextmenu_(e: PointerEvent) {
     const gesture = this.workspace.getGesture(e);
     if (gesture) {
       gesture.handleBlockStart(e, this);
+      gesture.handleRightClick(e);
```

Comment:
I was trying to figure out if it makes sense to call _both_ `handeBlockStart` and `handleRightClick`, given that the latter `.dispose()`es the gesture: it seemed pointless to start a gesture if we already know we're going to show the context menu.  But if I understand correctly it's necessary: `handleBlockStart` sets `gesture.targetBlock`, which is used by `handleRightClick`.

This seems convoluted, perhaps enough to warrant some additional commentary here if it can't be simplified.

Reply:
`handleRightClick` should be called from inside `doStart` in `gesture`, not directly from here. The gesture gets all of the information about what is going on with a click and decides what to do with it; the blocks and workspaces and fields just say that they've been hit.

`handleRightClick` is public, but only so that it could be called from `touch.ts:longStart`, and that is only called from inside `gesture:doStart`.

The code that checks if it should be a right click currently is:
```

    if (browserEvents.isRightButton(e)) {
      this.handleRightClick(e);
      return;
    }

    if (e.type.toLowerCase() === 'pointerdown' && e.pointerType !== 'mouse') {
      Touch.longStart(e, this);
    }

```

which happens from `handleBlockStart`. So a change to right-click handling should be done by changing that logic, not by adding an additional call into `handleRightClick` from outside of the gesture.